core/iwasm/common/wasm_c_api.c

```diff
@@ -340,6 +340,10 @@ wasm_engine_new_internal(mem_alloc_type_t type, const MemAllocOption *opts)
             opts->allocator.free_func;
         init_args.mem_alloc_option.allocator.realloc_func =
             opts->allocator.realloc_func;
+#ifdef MEM_ALLOC_WITH_USER_DATA
```

Comment:
Just out of curiosity, is there any reason (other than historical) for that pattern? I saw it's more common to use `ifndef/ifdef` pattern and the config file have all possible options listed, but commented if not used.

Reply:
Yes, the `ifndef/ifdef` pattern is also very common, but it may be not very convenience to use the config file: if it isn't created and all the configurations are controlled by makefile, it may be not easy to learn all the configs in the makefile. And if the config header file is used, (1) if it is auto generated e.g. autoconf.h, it may be under the output build folder and not easy to find it, (2) if it is one file of the project and all the configs are listed, then as you mentioned, if we want to enable/disable one config, we may need to uncomment/comment the macro manually or by makefile, just not so convenience as `#if xxx != 0`/`#if xxx == 0` pattern with which normally we don't need to modify the config file.